xmppserver/src/main/java/org/jivesoftware/openfire/OfflineMessageStore.java

```diff
@@ -110,6 +126,16 @@ public static OfflineMessageStore getInstance() {
     public OfflineMessageStore() {
         super("Offline Message Store");
         sizeCache = CacheFactory.createCache("Offline Message Size");
+        try
+        {
+            this.daystolive = JiveGlobals.getIntProperty("xmpp.offline.autoclean.daystolive", 14);
+            this.checkinterval = JiveGlobals.getIntProperty("xmpp.offline.autoclean.checkinterval", 30);
+            this.enableAutoClean = JiveGlobals.getBooleanProperty("xmpp.offline.autoclean.enabled", false);
```

Comment:
Please use the newer construct using `SystemProperty` instead of `JiveGlobals.getProperty`, for example

    public static final SystemProperty<Boolean> PUBSUB_FOOBAR = SystemProperty.Builder.ofType(Boolean.class)
        .setKey("xmpp.pubsub.create.FOOBAR")
        .setDefaultValue(false)
        .setDynamic(true)
        .build();

And then use it as follows:

    if (PUBSUB_FOOBAR.getValue()) { // ... 

This construct replaces the old JiveGlobals. It adds type-safety, and, more importantly, adds a good deal of automatic documentation / configuration options in the admin console.

Reply:
done...